### ISSUE TITLE
Body not known until after apply

### ISSUE THREAD

**Issue by kjagiello** [hashicorp/terraform-provider-http#101]
### Terraform Version

v1.0.11

### Affected Resource(s)

- data.http (2.1.0)

### Terraform Configuration Files

```hcl
data "http" "test" {
  url = "https://checkpoint-api.hashicorp.com/v1/check/terraform"
}
```

### Debug Output

```
  # module.test.data.http.test will be read during apply                            
  # (config refers to values not yet known)         
 <= data "http" "test"  {                                                                                
      + body             = (known after apply)                                                           
      + id               = (known after apply)                                                           
      + response_headers = (known after apply)
      + url              = "https://checkpoint-api.hashicorp.com/v1/check/terraform"                     
    }   
```

### Expected Behavior

I might be wrong here, but I expected the `body` of the response to be populated during Terraform's "refresh" phase, as per [docs](https://www.terraform.io/docs/language/data-sources/index.html#data-resource-behavior):

> If the query constraint arguments for a data resource refer only to constant values or values that are already known, the data resource will be read and its state updated during Terraform's "refresh" phase, which runs prior to creating a plan. This ensures that the retrieved data is available for use during planning and so Terraform's plan will show the actual values obtained.

### Background

I have stumbled upon this issue while trying to fetch a Kubernetes manifest from an external source. The manifest is a single YAML file that consists of multiple documents, so I'm splitting them up by the document marker (`---`) and then feed them into [`kubernetes_manifest`](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/manifest) using `for_each`.  See the module below for ference:

```hcl
data "http" "yaml_manifest" {
  url = var.url
}

locals {
  parsed_manifests = [
    for manifest in split("\n---\n", data.http.yaml_manifest.body) :
    yamldecode(manifest)
  ]
  named_manifests = {
    for manifest in local.parsed_manifests :
    join(":", [
      manifest.apiVersion,
      manifest.kind,
      manifest.metadata.name,
    ]) => manifest
  }
}


resource "kubernetes_manifest" "manifest" {
  for_each = local.named_manifests
  manifest = each.value
}
```

This results in a following error:

```
│ Error: Invalid for_each argument
│ 
│   on ../modules/utils/k8s_remote_manifest/main.tf line 22, in resource "kubernetes_manifest" "manifest":
│   22:   for_each = local.named_manifests
│     ├────────────────
│     │ local.named_manifests will be known only after apply
│ 
│ The "for_each" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created. To work around this, use the -target argument to
│ first apply only the resources that the for_each depends on.
```

So basically the request is not made during the "refresh" phase even though the URL for the datasource is known before apply. Is this an actual bug or am I simply misinterpreting the documentation?

**Comment by kjagiello** [comment]
I have realized that this is just the way all datasources work and that this is not a http-specific issue. Closing it.
